src/YieldDisburser.sol

```diff
@@ -53,19 +54,29 @@ contract YieldDisburser is OwnableUpgradeable {
         if (!_resolved) revert YieldNotResolved();
 
         breadToken.claimYield(breadToken.yieldAccrued(), address(this));
-
-        (uint256[] memory projectDistributions, uint256 totalVotes) = _getVotedDistribution(breadchainProjects.length);
-
-        lastClaimedTimestamp = Time.timestamp();
-        lastClaimedBlocknumber = Time.blockNumber();
+        uint256 breadchainProjectsLength = breadchainProjects.length;
+        (uint256[] memory projectDistributions, uint256 totalVotes) = _getVotedDistribution(breadchainProjectsLength);
+        if (totalVotes == 0) {
+            projectDistributions = new uint256[](breadchainProjectsLength);
+            for (uint256 i; i < breadchainProjectsLength; ++i) {
+                projectDistributions[i] = 1;
+            }
+            totalVotes = breadchainProjectsLength;
+        }
 
         uint256 halfBalance = breadToken.balanceOf(address(this)) / 2;
-        uint256 baseSplit = halfBalance / breadchainProjects.length;
+        uint256 baseSplit = halfBalance / breadchainProjectsLength;
 
-        for (uint256 i; i < breadchainProjects.length; ++i) {
+        for (uint256 i; i < breadchainProjectsLength; ++i) {
             uint256 votedSplit = ((projectDistributions[i] * halfBalance) / totalVotes);
             breadToken.transfer(breadchainProjects[i], votedSplit + baseSplit);
         }
+        if (queuedBreadchainProjects.length > 0) {
```

Comment:
this could be broken out into an internal function called like `_updateBreadchainProjects()`. This function is already pretty bloated with logic so I think any opportunity to reduce it and make it more readable should be taken

Reply:
addressed with 07c29b2 :)